File: Front-end/src/app/models/detalleMovimiento.ts
import { Producto } from "./producto";
import { Movimiento } from "./movimiento";

export class DetalleMovimiento {
  id!: number;
  cantidad!: number; // Cantidad del producto
  precioUnitario!: number; // Precio por unidad
  subtotal!: number; // Subtotal del detalle (cantidad * precioUnitario)
  movimiento!: Movimiento; // Relación con Movimiento (encabezado)
  producto!: Producto; // Relación con Producto
}

File: Back-end/src/main/java/com/springboot/backend/optica/modelo/DetalleMovimiento.java
package com.springboot.backend.optica.modelo;

import java.io.Serializable;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.Data;

@Entity
@Table(name = "detalle_movimientos")
@Data
public class DetalleMovimiento implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private int cantidad; // Cantidad del producto

    @Column(nullable = false)
    private float precioUnitario;

    @Column(nullable = false)
    private float subtotal;

    // Relación con el movimiento (encabezado)
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "movimiento_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Movimiento movimiento;

    // Relación con el producto
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "producto_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Producto producto;

    private static final long serialVersionUID = 1L;
}
